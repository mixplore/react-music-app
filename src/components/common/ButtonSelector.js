import React from "react";
import "./ButtonSelector.scss";

export default function ButtonSelector({ buttons, selectedButton, onButtonSelect }) {
  return (
    <div role="radiogroup" className="button-selector">
      {buttons.map((btn) => (
        <div key={btn.id}>
          <input
            id={btn.id}
            className="select-button"
            type="radio"
            name="select-option"
            value={btn.value}
            checked={selectedButton === btn.id}
            onChange={onButtonSelect}
          />
          <label htmlFor={btn.id}>{btn.title}</label>
        </div>
      ))}
    </div>
  );
}
