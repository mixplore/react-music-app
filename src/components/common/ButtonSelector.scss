.button-selector {
  display: flex;
  flex-direction: row;

  input[type="radio"] {
    visibility: hidden;
  }

  .select-button {
    + label {
      width: 6rem;
      height: 5rem;
      padding: 0.3rem;
      margin: 0 1rem 1rem 0;
      border-radius: 0.2rem;
      display: flex;
      align-items: center;
      justify-content: center;
      color: #fff;
      background-color: #d3d3d3;
      box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.1);
      cursor: pointer;
    }

    + label:hover {
      box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
    }

    &:checked + label {
      background-color: #791abe;
    }
  }
}
