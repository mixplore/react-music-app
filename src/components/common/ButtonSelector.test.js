import React from "react";
import renderer from "react-test-renderer";
import ButtonSelector from "./ButtonSelector";

describe(`#render`, () => {
  test("Snapshot test - Buttons are rendered correctly", () => {
    const buttons = [
      {
        id: "test-id1",
        value: "test-value1",
        title: "Test 1"
      },
      {
        id: "test-id2",
        value: "test-value2",
        title: "Test 2"
      },
    ];
    const selectedButton = "test-id1";
    const onButtonSelect = jest.fn();
    const component = renderer.create(
      <ButtonSelector
        buttons={buttons}
        selectedButton={selectedButton}
        onButtonSelect={onButtonSelect}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
