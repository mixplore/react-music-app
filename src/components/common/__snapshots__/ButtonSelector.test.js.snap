// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render Snapshot test - Buttons are rendered correctly 1`] = `
<div
  className="button-selector"
  role="radiogroup"
>
  <div>
    <input
      checked={true}
      className="select-button"
      id="test-id1"
      name="select-option"
      onChange={[MockFunction]}
      type="radio"
      value="test-value1"
    />
    <label
      htmlFor="test-id1"
    >
      Test 1
    </label>
  </div>
  <div>
    <input
      checked={false}
      className="select-button"
      id="test-id2"
      name="select-option"
      onChange={[MockFunction]}
      type="radio"
      value="test-value2"
    />
    <label
      htmlFor="test-id2"
    >
      Test 2
    </label>
  </div>
</div>
`;
